src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -261,64 +303,57 @@ public async Task<IEnumerable<WorkloadDownload>> GetManifestPackageDownloadsAsyn
 
         private IEnumerable<ManifestId> GetInstalledManifestIds() => _workloadResolver.GetInstalledManifests().Select(manifest => new ManifestId(manifest.Id));
 
-        private async Task UpdateAdvertisingManifestAsync(WorkloadManifestInfo manifest, bool includePreviews, DirectoryPath? offlineCache = null)
+        private async Task<bool> UpdateManifestWithVersionAsync(string id, bool includePreviews, SdkFeatureBand band, NuGetVersion packageVersion = null, DirectoryPath? offlineCache = null)
         {
+            var manifestId = new ManifestId(id);
             string packagePath = null;
-            var manifestId = new ManifestId(manifest.Id);
             try
             {
-                SdkFeatureBand? currentFeatureBand = null;
-                var fallbackFeatureBand = new SdkFeatureBand(manifest.ManifestFeatureBand);
-                // The bands should be checked in the order defined here.
-                SdkFeatureBand[] bands = [_sdkFeatureBand, fallbackFeatureBand];
-                var success = false;
-                // Use Distinct to eliminate bands that are the same.
-                foreach (var band in bands.Distinct())
+                var manifestPackageId = _workloadManifestInstaller.GetManifestPackageId(manifestId, band);
+                try
+                {
+                    // If an offline cache is present, use that. Otherwise, try to acquire the package online.
+                    packagePath = offlineCache != null ?
+                        Directory.GetFiles(offlineCache.Value.Value)
+                            .Where(path => path.EndsWith(".nupkg") && Path.GetFileName(path).StartsWith(manifestPackageId.ToString(), StringComparison.OrdinalIgnoreCase))
+                            .Max() :
```

Comment:
This looks like the correct logic if `packageVersion` is not specified.  If it is specified, I think this logic needs to take the version into account to return the package version specified instead of the latest version in the cache.

Reply:
That was how it was before...I'll admit I thought that was a little odd but figured it must not matter too much since we hadn't fixed it for other manifests? But I can change the logic for this.

Reply:
I think the difference is that previously it would always look for the latest version of a package, it didn't have the capability of specifying a specific version.

Reply:
Mmm...I think you're right. I'd thought there was something that handled versions previously, but I don't know what I was thinking of.